packages/core/src/fhirpath/functions.ts

```diff
@@ -34,7 +34,7 @@ export const functions: Record<string, FhirPathFunction> = {
    * @returns True if the input collection is empty ({ }) and false otherwise.
    */
   empty: (_context: AtomContext, input: TypedValue[]): TypedValue[] => {
-    return booleanToTypedValue(input.length === 0);
+    return booleanToTypedValue(input.length === 0 || input.every((e) => isEmpty(e.value)));
```

Comment:
This is effectively the same as the following:
```ts
    return booleanToTypedValue(input.every((e) => isEmpty(e.value)));
```

However, the current change is a bit more obvious IMO. Defer to maintainers on preference.